discovery/db_discovery.go

```diff
@@ -100,29 +121,60 @@ func (dbo *DBOrchestratorPoolCache) GetOrchestrators(numOrchestrators int) ([]*n
 }
 
 func (dbo *DBOrchestratorPoolCache) Size() int {
-	return len(dbo.GetURLs())
+	count, _ := dbo.store.OrchCount(
+		&common.DBOrchFilter{
+			MaxPrice:     server.BroadcastCfg.MaxPrice(),
+			CurrentRound: dbo.rm.LastInitializedRound(),
+		},
+	)
+	return count
 }
 
-func cacheRegisteredTranscoders(node *core.LivepeerNode) error {
-	orchestrators, err := node.Eth.RegisteredTranscoders()
+func (dbo *DBOrchestratorPoolCache) cacheTranscoderPool() error {
+	orchestrators, err := dbo.lpEth.TranscoderPool()
 	if err != nil {
-		glog.Error("Could not refresh DB list of orchestrators: ", err)
-		return err
+		return fmt.Errorf("Could not refresh DB list of orchestrators: %v", err)
 	}
-	_, dbOrchErr := cacheDBOrchs(node, orchestrators)
-	if dbOrchErr != nil {
-		glog.Error("Could not refresh DB list of orchestrators: cacheDBOrchs err")
+
+	for _, o := range orchestrators {
+		if err := dbo.store.UpdateOrch(ethOrchToDBOrch(o)); err != nil {
+			glog.Errorf("Unable to update orchestrator %v in DB: %v", o.ServiceURI, err)
```

Comment:
Since an orchestrator is identified by its ETH address in the DB can we use the ETH address instead of the ServiceURI here?

Reply:
fixed in e8acbd03